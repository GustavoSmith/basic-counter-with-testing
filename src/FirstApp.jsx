import React from "react";

const FirstApp = ({
  title = "No hay titulo...",
  subtitle = "No hay subtitulo...",
  name = "Gustavo",
}) => {
  return (
    <main className="flex h-screen flex-col items-center justify-center gap-16 bg-slate-800 text-white">
      <h1 className="text-5xl font-bold" data-testid="test-title"> {title} </h1>
      <h2 className="text-3xl font-semibold">{subtitle}</h2>
      <h2 className="text-3xl font-semibold">{subtitle}</h2>
      <h3 className="text-2xl">{name}</h3>
    </main>
  );
};

export default FirstApp;
