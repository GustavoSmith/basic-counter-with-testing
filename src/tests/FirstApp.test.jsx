import { render, screen } from "@testing-library/react";
import FirstApp from "../FirstApp";

describe("Pruebas en <FirstApp/>", () => {
  const title = "Soy el test";
  const subtitle = "Soy el subtitulo";

  test("Debe matchear con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el mensaje 'Soy el test'", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Debe mostrar el titulo en un h1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("Debe mostrar el subtitulo enviado por props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect(screen.getAllByText(subtitle)).toHaveLength(2);
  });
});
