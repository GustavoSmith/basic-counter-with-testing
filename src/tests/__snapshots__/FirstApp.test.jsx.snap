// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe matchear con el snapshot 1`] = `
<div>
  <main
    class="flex h-screen flex-col items-center justify-center gap-16 bg-slate-800 text-white"
  >
    <h1
      class="text-5xl font-bold"
      data-testid="test-title"
    >
       
      Soy el test
       
    </h1>
    <h2
      class="text-3xl font-semibold"
    >
      No hay subtitulo...
    </h2>
    <h2
      class="text-3xl font-semibold"
    >
      No hay subtitulo...
    </h2>
    <h3
      class="text-2xl"
    >
      Gustavo
    </h3>
  </main>
</div>
`;
